pkg/worker/model_status_update_job.go

```diff
@@ -33,9 +36,16 @@ func (w *Worker) ModelStatusUpdateJob(ctx context.Context, args ...interface{})
 
 	logger.Info("checking if model status should be updated, and creating notification")
 
+	// Ensure there's a context with the user account service
+	// TODO THis is a bit hacky -- we should probably be doing this on every job, not just this one, but we needed to get this working in MINT-3068
+	dataLoaders := loaders.NewDataLoaders(w.Store)
+	ctx = loaders.CTXWithLoaders(ctx, dataLoaders)
+	ctx = appcontext.WithUserAccountService(ctx, userhelpers.UserAccountGetByIDLOADER)
+
 	return resolvers.TrySendEmailForPhaseSuggestionByModelPlanID(
 		ctx,
 		w.Store,
+		appcontext.Principal(ctx),
```

Comment:
Since this is running in a Faktory job, I'm not actually sure what `appcontext.Principal(ctx)` will return here.

I think it might make sense to see if we can avoid touching the `modifiedDts`/`modifiedBy` fields here if possible, since, at least from a users/auditing perspective, we probably don't want to show this as an update being performed by someone.

Does the audit trigger complain if we don't set `modifiedDts`/`modifiedBy`? Perhaps we could add this column to the `ignored_fields` for this table?

Reply:
Yeah, running this locally seems to get a nil pointer dereference in the logs and the job panics (and retries) -- I think we might need to update this to work around needing to set those fields